contracts/strategies/donation-voting/DonationVotingStrategy.sol

```diff
@@ -0,0 +1,451 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.19;
+
+import {IAllo} from "../../core/IAllo.sol";
+import {IRegistry} from "../../core/IRegistry.sol";
+import {BaseStrategy} from "../BaseStrategy.sol";
+import {Metadata} from "../../core/libraries/Metadata.sol";
+import {ReentrancyGuard} from "@openzeppelin/security/ReentrancyGuard.sol";
+
+contract DonationVotingStrategy is BaseStrategy, ReentrancyGuard {
+    /// ================================
+    /// ========== Struct ==============
+    /// ================================
+
+    enum InternalRecipientStatus {
+        None,
+        Pending,
+        Accepted,
+        Rejected,
+        Appealed
+    }
+
+    /// @notice Struct to hold details of the recipients
+    struct Recipient {
+        bool isRegistryIdentity;
+        address recipientAddress;
+        InternalRecipientStatus recipientStatus;
+    }
+
+    /// @notice Struct to hold details of the allocations to claim
+    struct Claim {
+        address recipientId;
+        address token;
+    }
+
+    /// ===============================
+    /// ========== Errors =============
+    /// ===============================
+
+    error UNAUTHORIZED();
+    error RECIPIENT_ALREADY_ACCEPTED();
+    error REGISTRATION_NOT_ACTIVE();
+    error ALLOCATION_NOT_ACTIVE();
+    error ALLOCATION_NOT_ENDED();
+    error INVALID();
+    error DISTRIBUTION_STARTED();
+    error NOT_ALLOWED();
+
+    /// ===============================
+    /// ========== Events =============
+    /// ===============================
+
+    event Appealed(address indexed recipientId, bytes data, address sender);
+    event Reviewed(address indexed recipientId, InternalRecipientStatus recipientStatus, address sender);
+    event Claimed(address indexed recipientId, address recipientAddress, uint256 amount, address token);
+
+    /// ================================
+    /// ========== Storage =============
+    /// ================================
+
+    bool public registryGating;
+    bool public distributionStarted;
+    uint256 public registrationStartTime;
+    uint256 public registrationEndTime;
+    uint256 public allocationStartTime;
+    uint256 public allocationEndTime;
+    address[] public allowedTokens;
```

Comment:
`address[] public allowedTokens;` should be a mapping `address => bool`

Reply:
if it's empty, then it means all tokens are allowed. Not sure how we'd cover that scenario if we used a mapping
unless we say address(0) == true , means all tokens allowed

Reply:
i quite like `address(0) == true`

Reply:
done